python/ray/serve/air_integrations.py

```diff
@@ -57,19 +57,28 @@ def _load_predictor_cls(
     return predictor_cls
 
 
-def _unpack_tensorarray_from_pandas(output_df: "pd.DataFrame") -> "pd.DataFrame":
-    """Unpack predictor's return value with TensorArray into numpy.
+def _unpack_dataframe_to_serializable(output_df: "pd.DataFrame") -> "pd.DataFrame":
+    """Unpack predictor's pandas return value to JSON serializable format.
 
     In dl_predictor.py we return a pd.DataFrame that could have multiple
     columns but value of each column is a TensorArray. Flatten the
     TensorArray to list to ensure output is json serializable as http
     response.
+
+    In numpy predictor path, we might return collection of np.ndarrays that also
+    requires flattening to list to ensure output is json serializable as http
+    response.
     """
     from ray.data.extensions import TensorDtype
 
     for col in output_df.columns:
+        # TensorArray requires special handling to numpy array.
```

Comment:
actually, per our offline conversations, should we just remove all TensorArray stuff from Predictors?

Reply:
I'm assuming that we're leaving this relatively alone in this PR? Just double-checking, what was the decision?